Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -675,27 +762,42 @@ namespace OpenInfraPlatform {
 						(conversionMatrix._21 * vector3D.x + conversionMatrix._22 * vector3D.y + conversionMatrix._23 * vector3D.z));
 				}
 
-				// (5/6) IfcPolyline SUBTYPE OF IfcBoundedCurve
+				// IfcPolyline SUBTYPE OF IfcBoundedCurve
 				/*! \brief Converts \c IfcPolyline to a series of points.
 				 *
 				 * \param[in] ifcpolyline			The \c IfcPolyline to be converted.
 				 *
 				 * \returns							The series of points.
 				 */
-				std::vector<carve::geom::vector<3>> convertIfcPolyline(
-					const EXPRESSReference<typename IfcEntityTypesT::IfcPolyline>& ifcpolyline
+				void convertIfcPolyline(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcPolyline>& polyline, 
+					std::vector<carve::geom::vector<3>>& targetVec,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement
 				) const throw(...)
 				{
 					// **************************************************************************************************************************
-					// https://standards.buildingsmart.org/IFC/RELEASE/IFC4_1/FINAL/HTML/schema/ifcgeometryresource/lexical/ifcpolyline.htm
+					//	https://standards.buildingsmart.org/IFC/DEV/IFC4_3/RC2/HTML/schema/ifcgeometryresource/lexical/ifcpolyline.htm
+					//	ENTITY IfcPolyline
+					//		SUBTYPE OF(IfcBoundedCurve);
+					//		Points: LIST[2:? ] OF IfcCartesianPoint;
+					//	WHERE
+					//		SameDim : SIZEOF(QUERY(Temp < *Points | Temp.Dim <> Points[1].Dim)) = 0;
+					//	END_ENTITY;
 					// **************************************************************************************************************************
-					if (ifcpolyline.expired())
-						throw oip::ReferenceExpiredException(ifcpolyline);
+					if (polyline.expired())
+						throw oip::ReferenceExpiredException(polyline);
 
-					return convertIfcCartesianPointVector(ifcpolyline->Points);
+					std::vector<carve::geom::vector<3>> loop = convertIfcCartesianPointVector(polyline->Points);
+
+					// Internal TODO: Implement function GetPointOnCurve, which will be able calculate trimming for each curve
+					segmentStartPoints.push_back(loop.at(0));
+					targetVec.insert(targetVec.end(), loop.begin(), loop.end());

```

Comment:
Include a check, if `trim1Vec` or `trim2Vec` are not empty - throw a not supported exception.

Include this check in other functions that do not react to trimming stuff yet as well.

Reply:
Resolved: 475dbf1